18F/identity-idp
# LG-8026: Include error details in OTP confirmation submission

## 🎫 Ticket

[LG-8026](https://cm-jira.usa.gov/browse/LG-8026)

## 🛠 Summary of changes

Refactors `OtpVerificationForm` to use `ActiveModel` validations to record specific validation issues with an OTP verification attempt, in order to satisfy requirements of LG-8026 to have queryable insight to expired token submissions.

## 📜 Testing Plan

1. With IdP running, run logging observer in a separate terminal process: `tail -F -n0 log/events.log | jq`
2. Sign in with an account configured to use phone MFA
3. (If you're not prompted to confirm MFA, click "Forget passwords" from account dashboard, sign out, then repeat step 2)
4. Enter the incorrect code
5. Observe that logging details include `error_details` for specific validation issues
